### PR TITLE
Fix API when query by entrezGeneid

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

File: core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java [hunk 1]
```diff
@@ -47,25 +47,27 @@ public void testProcessQuery() throws Exception {
...
+        query = new Query(null, null, null, "BRAF", "V600E", null, "Breast Cancer", null, null, null);
         indicatorQueryResp = IndicatorUtils.processQuery(query, null, null, null, true);
         assertEquals("The highest sensitive level should be 2B", LevelOfEvidence.LEVEL_2B, indicatorQueryResp.getHighestSensitiveLevel());
         assertTrue("The highest resistance level should be null", indicatorQueryResp.getHighestResistanceLevel() == null);
         assertTrue("Shouldn't have any significant level", indicatorQueryResp.getOtherSignificantSensitiveLevels().size() == 0);
         assertTrue(treatmentsContainLevel(indicatorQueryResp.getTreatments(), LevelOfEvidence.LEVEL_2B));
 
         // Test for predicted oncogenic
-        query = new Query("KRAS", null, "\tQ61K", null, "Pancreatic Adenocarcinoma", null, null, null);
+        query = new Query(null, null, null, "KRAS", "Q61K", null, "Pancreatic Adenocarcinoma", null, null, null);
```
Comment (JiaoJiao123) [inline core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java]: don't you want to test on "\tQ61K"? You were testing that previously.

File: core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java [hunk 2]
```diff
@@ -103,14 +105,49 @@ public void testProcessQuery() throws Exception {
...
+        assertTrue("Treatments are not the same", resp1.getTreatments().equals(resp2.getTreatments()));
+        assertTrue("Highest sensitive levels are not the same.", LevelUtils.areSameLevels(resp1.getHighestSensitiveLevel(), resp2.getHighestSensitiveLevel()));
+        assertTrue("Highest resistance levels are not the same.", LevelUtils.areSameLevels(resp1.getHighestResistanceLevel(), resp2.getHighestResistanceLevel()));
+
+        // Check hugoSymbol and entrezGene pair
+        query1 = new Query(null, null, 673, null, "V600E", null, "Melanoma", null, null, null);
+        query2 = new Query(null, null, null, "BRAF", "V600E", null, "Melanoma", null, null, null);
+        resp1 = IndicatorUtils.processQuery(query1, null, null, null, true);
+        resp2 = IndicatorUtils.processQuery(query2, null, null, null, true);
+        assertTrue("Genes are the same", resp1.getGeneSummary().equals(resp2.getGeneSummary()));
```
Comment (JiaoJiao123) [inline core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java]: The message should be "Genes are not the same", right? same with 2 other assertTrue() below.

File: core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java (hunk 1 above)
Comment (zhx828) [inline core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java]: It was a typo

File: core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java (hunk 2 above)
Comment (zhx828) [inline core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java]: Updated.

**Pull request merged**